diagnose_cluster.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python
+
+import argparse
+import re
+import json
+from subprocess import call, Popen, PIPE
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--name', '-n', required=True, type=str, help="Cluster name")
+parser.add_argument('--dest', '-d', required=True, type=str, help="Directory for diagnose output -- must be local")
+parser.add_argument('--hail-log', '-l', required=False, type=str, default='/home/hail/hail.log', help="Path for hail.log file")
+parser.add_argument('--overwrite', required=False, action='store_true', help="Delete dest directory before adding new files")
+parser.add_argument('--no-diagnose', required=False, action='store_true', help="Do not run gcloud dataproc clusters diagnose")
+parser.add_argument('--compress', '-z', required=False, action='store_true', help="GZIP all files")
+parser.add_argument('--workers', required=False, nargs='*', help="Specific workers to get log files from")
+parser.add_argument('--take', required=False, type=int, default=None, help="Only download logs from the first N workers")
+
+
+args = parser.parse_args()
+
+is_local = not args.dest.startswith("gs://")
+
+if args.overwrite:
+    if is_local:
+        call('rm -r {dir}'.format(dir=args.dest), shell=True)
+    else:
+        call('gsutil -m rm -r {dir}'.format(dir=args.dest), shell=True)
+    
+
+master_dest = args.dest.rstrip('/') + "/master/"
+worker_dest = args.dest.rstrip('/') + "/workers/"
+
+if is_local:
+    call('mkdir -p {dir}'.format(dir=master_dest), shell=True)
+    call('mkdir -p {dir}'.format(dir=worker_dest), shell=True)
+
+desc = json.loads(Popen('gcloud dataproc clusters describe {name} --format json'.format(name=args.name),
```

Comment:
Oof. I know I'm a PITA about this, but this exact line is why you don't want to pass shell=True:

```
CLUSTER="; rm -rf /"  # not saying you would do this but crazy shit happens in the shell
python diagnose_cluster.py --name $CLUSTER
# oops
```

The party-line about this is that `shell=True` should be avoided at all costs. Instead, create a list with the exact arguments you want, and then call `subprocess.check_output`